src/App.tsx

```diff
@@ -1,22 +1,32 @@
 import * as React from 'react';
+import { BrowserRouter as Router, Route, Redirect } from 'react-router-dom';
+import SecuredRoute from './components/SecuredRoute';
+
 import './App.scss';
 import './assets/styles/common.scss';
 
 import { MainLayout } from './components/Layouts/MainLayout';
+import { Authorization } from './pages/Authorization';
 
 import { getAsset } from './services/asset';
 import { IAsset } from './interfaces/asset';
 import { observer, inject } from 'mobx-react';
-import { UserSettingsModal } from './components/UserSettingsModal';
 import { host } from './env/environment';
 
 interface IAppProps {
   stores?: any;
 }
 
+interface IAppState {}
+
 @inject('stores')
 @observer
-export default class App extends React.Component<IAppProps, {}> {
+export default class App extends React.Component<IAppProps, IAppState> {
+  constructor(props: IAppProps) {
+    super(props);
+    this.props.stores.userDetailsStore.user = JSON.parse(localStorage.getItem('userDetails'));
```

Comment:
to keep it more readable :
const userDetails = JSON.parse(localStorage.getItem('userDetails'));
this.props.stores.userDetailsStore.user = userDetails